internal/command/apply.go

```diff
@@ -56,6 +56,11 @@ func (c *ApplyCommand) Run(rawArgs []string) int {
 		return 1
 	}
 
+  if diags.HasWarnings() {
+    view.Diagnostics(diags)
+    diags = nil
+  }
+
```

Comment:
I don't think we need this, the warning should still be printed whenever the diagnostics are chosen to be delayed.

Is there something else that is dropping / ignoring the warning diagnostics later?

Reply:
Actually, it is required to print the diagnostics before, since on line number 73 `planFile, diags := c.LoadPlanFile(args.PlanPath)` returns a new diagnostic slice, it doesn't append to the already available `diags` variable.

Reply:
Cool, I think we can change the other part instead of doing this 👍 It's more in keeping with the way the diagnostics are intended to work to just gather them up and report once.


You can change line 73 to something like:

```
	planFile, moreDiags := c.LoadPlanFile(args.PlanPath)
        diags = diags.Append(moreDiags)
	if moreDiags.HasErrors() {
		view.Diagnostics(diags)
		return 1
	}
```

Reply:
> Cool, I think we can change the other part instead of doing this 👍 It's more in keeping with the way the diagnostics are intended to work to just gather them up and report once.
> 
> You can change line 73 to something like:
> 
> ```
> 	planFile, moreDiags := c.LoadPlanFile(args.PlanPath)
>         diags = diags.Append(moreDiags)
> 	if moreDiags.HasErrors() {
> 		view.Diagnostics(diags)
> 		return 1
> 	}
> ```

Yeah, this can be done.